### ISSUE TITLE
Dropout not disabled for eval when using SDPA

### ISSUE THREAD

**Issue by sailordiary** [wenet-e2e/wenet#2699]
**Describe the bug**
Calls to torch SDPA are not passing correct dropout rates for eval (the functional version is used) -- should be 0 if not in training mode. This seems like a major issue. See, for instance,

https://github.com/wenet-e2e/wenet/blob/1a94771df418325fd335b0b53776f83d74a8111d/wenet/transformer/attention.py#L298
https://github.com/wenet-e2e/wenet/blob/1a94771df418325fd335b0b53776f83d74a8111d/wenet/firered/attention.py#L176

**Expected behavior**
According to https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html:
```
This function always applies dropout according to the specified dropout_p argument. To disable dropout during evaluation, be sure to pass a value of 0.0 when the module that makes the function call is not in training mode.
```

**Additional context**
Should check all calls to torch SDPA.

**Comment by Mddct** [comment]
pr welcome。
